stub_uploader/metadata.py

```diff
@@ -112,6 +112,10 @@ def requires_python(self) -> str | None:
         verify_requires_python(req)
         return req
 
+    @property
+    def upstream_repository(self) -> str | None:
+        return self.data.get("upstream_repository")
```

Comment:
Possibly we could do some validation here. Typeshed already does [a bunch of validation checks](https://github.com/python/typeshed/blob/54a8fd74a785ff7916bade3699a2be4542f1b1c0/tests/parse_metadata.py#L222-L244) for this field, though, so maybe it's not necessary.

Reply:
A few general notes: Generally the stub_uploader should treat everything coming from typeshed as "tainted" as a precaution against an attacker gaining access to a typeshed maintainer's GitHub account. (This is the reason why we have so few stub_uploader maintainers -> to limit the attack vector. Personally, I consider every typeshed maintainer to be a stub_uploader maintainer, just without the commit rights.) Crashing the uploader is fine, the risk is uploading potentially harmful code under a "types-" name.

That said, an `isinstance` check and regexp with "safe" markdown chars can't hurt here. The regexp check should probably just print a warning and return `None`.

Reply:
I added a little more validation in https://github.com/typeshed-internal/stub_uploader/pull/105/commits/796d2ea3bd5e9365a93c6e60287011b022a3774d. We could add more. Should we return `None` if `.isascii()` returns `False`? If a regex, what regex?

Open to ideas here, not sure how far we should go :)